zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbMappingState.java

```diff
@@ -34,38 +34,52 @@ public class DbMappingState implements MutableMappingState {
   private final ColumnFamily<DbLong, DbForeignKey<DbCompositeKey<DbString, DbString>>>
       claimByKeyColumnFamily;
 
+  private final DbString mappingId;
+  private final ColumnFamily<DbString, DbForeignKey<DbCompositeKey<DbString, DbString>>>
+      claimByIdColumnFamily;
+
   public DbMappingState(
       final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
     claimName = new DbString();
     claimValue = new DbString();
     claim = new DbCompositeKey<>(claimName, claimValue);
+    final PersistedMapping persistedMapping = new PersistedMapping();
     mappingColumnFamily =
         zeebeDb.createColumnFamily(
-            ZbColumnFamilies.MAPPINGS, transactionContext, claim, new PersistedMapping());
+            ZbColumnFamilies.MAPPINGS, transactionContext, claim, persistedMapping);
 
     mappingKey = new DbLong();
     fkClaim = new DbForeignKey<>(claim, ZbColumnFamilies.MAPPINGS);
     claimByKeyColumnFamily =
         zeebeDb.createColumnFamily(
             ZbColumnFamilies.CLAIM_BY_KEY, transactionContext, mappingKey, fkClaim);
+
+    mappingId = new DbString();
+    claimByIdColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.CLAIM_BY_KEY, transactionContext, mappingId, fkClaim);
   }
 
   @Override
   public void create(final MappingRecord mappingRecord) {
```

Comment:
❓ I assume you will deal with the other methods (delete/get/update) in different issues?